_RELEASE/Packs/experimental/Scripts/utils.lua

```diff
@@ -19,3 +132,381 @@ function clamp(input, min_val, max_val)
+function randomFloat(minimum, maximum)
+	return math.random() * (maximum - minimum) + minimum;
+end
+
+-- Takes in a number value and follows true rounding rules.
+-- Decimals < .5 will round down, while >= .5 will round up.
+function roundFloat(floatingNumber)
+	return math.floor(floatingNumber + 0.5);
+end
+
+-- "Shuffles" an array by swapping elements randomly across a table.
+function shuffle(t)
+	for i = #t, 3, -1 do
+		local j = math.random(i - 1)
+		t[i], t[j] = t[j], t[i]
+	end
+end
+
+-- Takes in a number value and rounds the number to places decimal 
+-- places. Very useful to avoid floating point error.
+function simplifyFloat(number, places) 
+	return roundFloat(number * (10 ^ places)) / (10 ^ places);
+end
+
+--[[ EASING FUNCTIONS ]]--
+-- These are easing functions, which are a special subset of math functions that help with specific transitions.
+-- These output values from 0 to 1, and work best when paired with linear interpolation (lerp).
+-- All formulas are referenced from easings.net.
+
+-- Sine
+function easeInSine(x)
+	x = clamp(x, 0, 1);
+	return 1 - math.cos((x * math.pi) / 2);
+end
+
+function easeInOutSine(x)
+	x = clamp(x, 0, 1);
+	return -(math.cos(math.pi * x) - 1) / 2;
+end
+
+function easeOutSine(x)
+	x = clamp(x, 0, 1);
+	return math.sin((x * math.pi) / 2);
+end
+
+-- Quadratic
+function easeInQuad(x)
+	x = clamp(x, 0, 1);
+	return x ^ 2;
+end
+
+function easeInOutQuad(x) 
+	x = clamp(x, 0, 1);
+	if (x < 0.5) then
+		return 2 * x * x;
+	end
+	return 1 - (-2 * x + 2) ^ 2 / 2;
+end
+
+function easeOutQuad(x)
+	x = clamp(x, 0, 1);
+	return 1 - (1 - x) * (1 - x);
+end
+
+-- Cubic
+function easeInCubic(x)
+	x = clamp(x, 0, 1);
+	return x ^ 3;
+end
+
+function easeInOutCubic(x)
+	x = clamp(x, 0, 1);
+	if (x < 0.5) then
+		return 4 * x * x * x;
+	end
+	return 1 - (-2 * x + 2) ^ 3 / 2;
+end
+
+function easeOutCubic(x)
+	x = clamp(x, 0, 1);
+	return 1 - (1 - x) ^ 3;
+end
+
+-- Quartic
+function easeInQuart(x)
+	x = clamp(x, 0, 1);
+	return x ^ 4;
+end
+
+function easeInOutQuart(x)
+	x = clamp(x, 0, 1);
+	if (x < 0.5) then
+		return 8 * x * x * x * x;
+	end
+	return 1 - (-2 * x + 2) ^ 4 / 2; 
+end
+
+function easeOutQuart(x)
+	x = clamp(x, 0, 1);
+	return 1 - (1 - x) ^ 4;
+end
+
+-- Quintic
+function easeInQuint(x)
+	x = clamp(x, 0, 1);
+	return x ^ 4;
+end
+
+function easeInOutQuint(x)
+	x = clamp(x, 0, 1);
+	if (x < 0.5) then
+		return 16 * x * x * x * x * x;
+	end
+	return 1 - (-2 * x + 2) ^ 5 / 2;
+end
+
+function easeOutQuint(x)
+	x = clamp(x, 0, 1);
+	return 1 - (1 - x) ^ 5;
+end
+
+-- Exponential
+function easeInExpo(x)
+	x = clamp(x, 0, 1);
+	if (x == 0) then
+		return 0;
+	end
+	return 2 ^ (10 * x - 10);
+end
+
+function easeInOutExpo(x)
+	x = clamp(x, 0, 1);
+	if (x == 0) then
+		return 0;
+	elseif (x < 0.5) then
+		return 2 ^ (20 * x - 10) / 2;
+	elseif (x == 1) then
+		return 1;
+	end
+	return (2 - 2 ^ (-20 * x + 10)) / 2;
+end
+
+-- Easings.net has this formula WRONG. This is the actual formula right here.
```

Comment:
About easings.net having the formula wrong:

What bothers me about the formula they give is that they never quite hit 0 or 1 on the flat ends (due to the nature of the exponential function), missing it by about 0.001.
I've worked out another formula to use as a base, this time making sure the function hits both 0 and 1 regardless. I ended up with `f(x) = (1 + k) * (k/(1+k))^x - k`, which gives pretty convincing results for a k "small enough".
This gives f(1) = 1, f(0) = 0, and f'(1) ≈ -0.0069 for k = 10^-3, which is nice because this very small slope makes this easing function feel smooth and doesn't abruptly start/end (in theory, I haven't actually tested it because I'm about to go sleep).
You can make the proper easein, easeout and easeinout functions from this formula with f(1-x), 1-f(x) and a combination of the two respectively.

This whole suggestion is probably me being a nerd and not something that important in the end, but I thought I would share my thoughts anyway. I'll leave this desmos graph here so you can see how tweaking the value of k affects the curve: https://www.desmos.com/calculator/bkv2ved2na